lib/addons/secrets-store/csi-driver-provider-aws-secrets.ts

```diff
@@ -0,0 +1,195 @@
+import { ClusterInfo } from '../../../lib';
+import { ApplicationTeam } from '../../teams';
+import { CfnOutput, Construct } from '@aws-cdk/core';
+import { ISecret } from '@aws-cdk/aws-secretsmanager';
+import { IStringParameter } from '@aws-cdk/aws-ssm';
+import { SecretProvider } from './secret-provider';
+
+/**
+ * TeamSecret Props
+ */
+export interface TeamSecretsProps {
+  secretProvider: SecretProvider;
+  kubernetesSecret?: KubernetesSecret;
+}
+
+/**
+ * Configuration for Kubernetes Secrets
+ */
+export interface KubernetesSecret {
+
+  /**
+   * Kubernetes Secret Name
+   */
+  secretName: string;
+
+  /**
+   * Type of Kubernetes Secret
+   */
+  type?: KubernetesSecretType
+
+  /**
+   * Secret Labels
+   */
+  labels?: Map<string, string>;
+
+  /**
+   * Kubernetes SecretObject Data
+   */
+  data?: KubernetesSecretObjectData[];
+}
+
+/**
+ * Data for Kubernetes Secrets
+ */
+interface KubernetesSecretObjectData {
+
+  /**
+   * Name of the AWS Secret that is syncd
+   */
+  objectName?: string;
+
+  /**
+   * Kubernetes Secret Key
+   */
+  key?: string;
+}
+
+enum AwsSecretType {
+  SSMPARAMETER = 'ssmparameter',
+  SECRETSMANAGER = 'secretsmanager'
+}
+
+export enum KubernetesSecretType {
+  OPAQUE = 'Opaque',
+  BASIC_AUTH = 'kubernetes.io/basic-auth',
+  TOKEN = 'bootstrap.kubernetes.io/token',
+  DOCKER_CONFIG_JSON = 'kubernetes.io/dockerconfigjson',
+  DOCKER_CONFIG = 'kubernetes.io/dockercfg',
+  SSH_AUTH = 'kubernetes.io/ssh-auth',
+  SERVICE_ACCOUNT_TOKEN = 'kubernetes.io/service-account-token',
+  TLS = 'kubernetes.io/tls'
+}
+
+interface parameterObject {
+    objectName: string;
+    objectType: string;
```

Comment:
The objects field of the SecretProviderClass can contain the following sub-fields:
* objectName: This field is required. It specifies the name of the secret or parameter to be fetched. For Secrets Manager this is the [SecretId](https://docs.aws.amazon.com/secretsmanager/latest/apireference/API_GetSecretValue.html#API_GetSecretValue_RequestParameters) parameter and can be either the friendly name or or full ARN of the secret. For SSM Parameter Store, this must be the [Name](https://docs.aws.amazon.com/systems-manager/latest/APIReference/API_GetParameter.html#API_GetParameter_RequestParameters) of the parameter and can not be a full ARN.
* objectType: This field is optional when using a Secrets Manager ARN for objectName, otherwise it is required. This field can be either "secretsmanager" or "ssmparameter".
* objectAlias: This optional field specifies the file name under which the secret will be mounted. When not specified the file name defaults to objectName.
* objectVersion: This field is optional, and generally not recommended since updates to the secret require updating this field. For Secrets Manager this is the [VersionId](https://docs.aws.amazon.com/secretsmanager/latest/apireference/API_GetSecretValue.html#API_GetSecretValue_RequestParameters). For SSM Parameter Store, this is the optional [version number](https://docs.aws.amazon.com/systems-manager/latest/userguide/sysman-paramstore-versions.html#reference-parameter-version).
* objectVersionLabel: This optional fields specifies the alias used for the version. Most applications should not use this field since the most recent version of the secret is used by default. For Secrets Manager this is the [VersionStage](https://docs.aws.amazon.com/secretsmanager/latest/apireference/API_GetSecretValue.html#API_GetSecretValue_RequestParameters). For SSM Parameter Store this is the optional [Parameter Label](https://docs.amazonaws.cn/en_us/systems-manager/latest/userguide/sysman-paramstore-labels.html).